src/ds/d500/d500-device.cpp

```diff
@@ -94,8 +94,35 @@ namespace librealsense
     void d500_device::hardware_reset()
     {
         command cmd(ds::HWRST);
-        cmd.require_response = false;
-        _hw_monitor->send(cmd);
+        cmd.require_response = true;
+        int retries = 0;
+        bool success = false;
+        const int ERR_SWNotReady = -21;
+        const int ERR_Operation_Timeout = -18;
+        do
+        {
+            auto res = _hw_monitor->send(cmd);
+            if (res.size() == 0)
+                return; // happens when no error occurs
+
+            auto hwm_ans = *reinterpret_cast<hwmon_response*>(res.data());
+            if (hwm_ans == hwmon_response::hwm_Success)
```

Comment:
So some D500 devices will return 0 data and some success opcode?
We need to ask FW team to align this behavior
both should return success IMO